doc/doxygen/riot.doxyfile

```diff
@@ -32,7 +32,7 @@ DOXYFILE_ENCODING      = UTF-8
 # title of most generated pages and in a few other places.
 # The default value is: My Project.
 
-PROJECT_NAME           =
+PROJECT_NAME           = Documentation
```

Comment:
```suggestion
PROJECT_NAME           = "RIOT Documentation"
```

Reply:
If I'm not mistaken, this change is made on purpose: "This PR changes the project name from "RIOT-Documentation" to "Documentation" from riot.doxyfile because from the main page of RIOT website, it is understood the documentation is about RIOT. "
And I agree with that.

Reply:
Mh, I prefer it the way it used to be (as documented in https://github.com/RIOT-OS/RIOT/issues/10815). While it is true that it is most likely understood that the documentation is about RIOT, it feels weird to call the "project" just documentation. Also, if doxygen at some point supports multiple projects (or combining multiple projects) we are future proof ;-).

Reply:
> as documented in #10815

I can't find there where a proposal about the use of "RIOT Documentation" over something else is documented.
To me it makes more sense to name the project "Documentation". It's clear that it's about RIOT.

> we are future proof

I consider this as premature optimization and it's very unlikely to happen.

Reply:
To me `Documentation` as a project name in doxygen doesn't make sense either, bc what other than documentation is doxygen about? That in mind, a proper project name is `RIOT-OS`.

Reply:
also note that, the relation to the riot-os webpage is only one possible way of publishing this. We could also think about exporting this to readthedocs (I think there are ways to generate the proper format from doxygen) and then a project name `Documentation` would make no sense at all, but `RIOT-OS` would.

Reply:
> That in mind, a proper project name is RIOT-OS

If you look at the variable name in the code, this is true. But if you look at the rendered documentation with CSS, top banner, etc then it's just redundant information.

Reply:
Regarding readthedocs or multi project management, this is out of scope here. This can be reconsidered later if a decision is taken towards one of these directions.

Reply:
still `Documentation` as a project names makes no sense to me and is also redundant bc what else is doxygen doing?

Also note that the project name is the tab-name in most browser. Assuming other projects would follow your argumentation, and you open all of them in separate tabs in your browser you will have N tabs with title `Documentation` so how will you then find the RIOT-OS documentation again? Wouldn't it be better to have a tab with `RIOT-OS` as its title to quickly find and click on it?

So I'm still in favour of  a simple `RIOT-OS` for project name,

Reply:
> Regarding readthedocs or multi project management, this is out of scope here. This can be reconsidered later if a decision is taken towards one of these directions.

Why enforce a rename later down the road if we can prevent it now?

Reply:
> Why enforce a rename later down the road if we can prevent it now?

Because now it makes sense. Nobody knows if a rename will be needed later.

> So I'm still in favour of a simple RIOT-OS for project name,

This makes sense regarding the tab name. I would just use a plain `RIOT`.

Reply:
> Because now it makes sense. Nobody knows if a rename will be needed later.

But thats just it: I argue (and I think @smlng agrees) that the current name *doesn't* make sense.

Reply:
> This makes sense regarding the tab name. I would just use a plain RIOT.

Then lets just use `RIOT`?

Reply:
I'm still more in favour of `RIOT-OS` bc it's non-ambiguous; there is also `riot.im` ... just sayin 😉

Anyway, wouldn't block plain `RIOT` either.

Reply:
IMHO, the idea of this PR was to give a good usability of the documentation stuff to the reader as shown in https://github.com/RIOT-OS/RIOT/issues/10815#issuecomment-572288741 and it should not be dependent/compromised by the doxygen limited features. Maybe, can give the project name as suggested above but usability is also important.



Possible solutions in my mind:
1) Choice of a better platform rather doxygen
2) Rely on upcoming features in doxygen (https://stackoverflow.com/questions/13368350/use-the-readme-md-file-as-main-page-in-doxygen)

Reply:
I think both solutions are rather long-term solutions. I think making `PROJECT_NAME` equal "RIOT Documentation" or "RIOT-OS Documentation" or "RIOT" or "RIOT-OS" makes it already much more usable. "Documentation" seems lacking information to me.

Reply:
To me we should view this from the tool perspective, i.e., doxygen here, and not how we use its (HTML) output and integrate it somehow into the riot-os.org webpage. Further, the specific variable `PROJECT NAME` is used by doxygen internally in many places (I can only stress this point again), e.g. for HTML output as the browser tab title. Hence this variable should have an overall (generic) meaningful value and not just something that is best for the one usage discussed here, i.e., integration in riot-os webpage. 

Again, IMHO that would be `RIOT-OS`.

Reply:
As I discussed before https://github.com/RIOT-OS/RIOT/pull/13064#discussion_r366432613 . I have no problem with project name (RIOT, RIOT-OS). My point is to have usability something as shown in figure.
![RIOT](https://user-images.githubusercontent.com/53410159/72534650-74333c80-3877-11ea-8893-46cac5ff496a.png)
  (where "v" represent open tab and ">" closed tab)
And i guess doxygen currently does not support that. i.e creating .md files with (tab, sub-tab, subsub-tab) and adding them to the main page just like Share-Latex. and for that either we have to wait for mainpage further supporting features.
I am thinking about long term solutions. No doubt currently, we can manage temporarily somehow but I guess that is not permanent solution.

Reply:
I think what @smlng meant was *browser* tabs. With a very recent `doxygen` (v1.8.17) the tree current version of this PR looks like this

![image](https://user-images.githubusercontent.com/675644/73283064-d3d40500-41f2-11ea-94ec-a1c810b4030b.png)

So RIOT does not appear at all and the tab's title is just "Overview".

On current master it looks like this

![image](https://user-images.githubusercontent.com/675644/73283320-388f5f80-41f3-11ea-8b05-a88ae5bf6702.png)

and the tab's title is "RIOT Documentation".

For older Doxygen versions see https://doc.riot-os.org/.